### PR TITLE
feat(Core): replace boost::filesystem with std::filesystem

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
## Changes Proposed:

Since C++17, the `boost::filesystem` has been included into the Standard C++ (see [Filesystem library](https://en.cppreference.com/w/cpp/filesystem)) so there is no reason to use Boost for the filesystem anymore.

## Tests Performed:

- Run `worldserver` process (macOS) on older AC databases and it correctly updated all of them:

```
 Updating Auth database...
>> Reapplying update "2021_05_13_00.sql" 'A502A30' -> 'B9CABD6' (it changed)...
>> Applied 1 query. Containing 7 new and 0 archived updates.
 
Updating Character database...
>> Applying update "2021_07_06_00.sql" 'E2B9577'...
>> Applied 1 query. Containing 12 new and 0 archived updates.
 
Updating World database...
>> Applying update "2021_07_03_02.sql" '34768C3'...
>> Applying update "2021_07_04_00.sql" '64E6B1C'...
>> Applying update "2021_07_04_01.sql" '1ECFD15'...
>> Applying update "2021_07_04_02.sql" '25F4D37'...
>> Applying update "2021_07_04_03.sql" 'DEE36BC'...
>> Applying update "2021_07_05_00.sql" '1658DE7'...
>> Applying update "2021_07_05_01.sql" '1DA606C'...
>> Applying update "2021_07_05_02.sql" '7D69C40'...
>> Applying update "2021_07_05_03.sql" 'A73D778'...
>> Applying update "2021_07_05_04.sql" '50C3B88'...
>> Applying update "2021_07_05_05.sql" '8F147CC'...
>> Applying update "2021_07_05_06.sql" 'CD17698'...
>> Applying update "2021_07_05_07.sql" 'C3A6A90'...
>> Applying update "2021_07_06_00.sql" '91BACF8'...
>> Applying update "2021_07_06_01.sql" 'BDC59D2'...
>> Applying update "2021_07_06_02.sql" '8F4F46E'...
>> Applying update "2021_07_06_03.sql" '8822BA2'...
>> Applying update "2021_07_06_04.sql" '01EC0B5'...
>> Applying update "2021_07_07_00.sql" '7EAF33F'...
>> Applying update "2021_07_07_01.sql" '69A671A'...
>> Applying update "2021_07_07_02.sql" '2A804C7'...
>> Applying update "2021_07_07_03.sql" 'F5AA6D4'...
>> Applying update "2021_07_07_04.sql" '2C7CB21'...
>> Applying update "2021_07_07_05.sql" 'E006EBF'...
>> Applying update "2021_07_07_06.sql" 'E7388D8'...
>> Applying update "2021_07_07_07.sql" 'FC21D56'...
>> Applying update "2021_07_07_08.sql" 'F9D7D75'...
>> Applying update "2021_07_07_09.sql" '1726119'...
>> Applying update "2021_07_07_10.sql" '6483574'...
>> Applying update "2021_07_07_11.sql" 'FCC7D84'...
>> Applying update "2021_07_07_12.sql" 'D0EE617'...
>> Applying update "2021_07_07_13.sql" 'B3ED8F5'...
>> Applying update "2021_07_07_14.sql" 'E74EFC1'...
>> Applying update "2021_07_07_15.sql" 'EE98A4D'...
>> Applying update "2021_07_07_16.sql" '01025B8'...
>> Applying update "2021_07_07_17.sql" '4569724'...
>> Applying update "2021_07_07_18.sql" '34C3B6B'...
>> Applying update "2021_07_07_19.sql" '893EF6D'...
>> Applying update "2021_07_07_20.sql" 'BB8CC8B'...
>> Applying update "2021_07_07_21.sql" '08B0336'...
>> Applying update "2021_07_07_22.sql" '6A700BD'...
>> Applied 41 queries. Containing 570 new and 0 archived updates.
```

## How to Test the Changes:

1. Test the DB updater inside the core

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/database/Updater/UpdateFetcher.cpp [hunk 1]
```diff
@@ -4,19 +4,17 @@
  */
 
 #include "UpdateFetcher.h"
-#include "Common.h"
```
Comment (Helias) [inline src/server/database/Updater/UpdateFetcher.cpp:7]: ?

File: src/server/database/Updater/UpdateFetcher.cpp [hunk 2]
```diff
@@ -4,19 +4,17 @@
...
 #include "QueryResult.h"
 #include "Tokenize.h"
 #include "Util.h"
-#include <boost/filesystem/operations.hpp>
-#include <fstream>
 #include <sstream>
+#include <fstream>
 
-using namespace boost::filesystem;
+using namespace std::filesystem;
```
Comment (Helias) [inline src/server/database/Updater/UpdateFetcher.cpp:17]: is not needed to include the filesystem library?

File: src/server/database/Updater/UpdateFetcher.cpp (hunk 1 above)
Comment (FrancescoBorzi) [reply src/server/database/Updater/UpdateFetcher.cpp:7]: unused import

File: src/server/database/Updater/UpdateFetcher.cpp (hunk 2 above)
Comment (FrancescoBorzi) [reply src/server/database/Updater/UpdateFetcher.cpp:17]: this file include `UpdateFetcher.h` which imports `<filesystem>`

**Review by Helias — APPROVED** [review]

**Comment by Kitzunu** [comment]
I will test tonight

**Review by Winfidonarleyan — APPROVED** [review]

**Review by Kitzunu — APPROVED** [review]

**Comment by Meramaf** [comment]
Don't work on Debian 10

**Comment by FrancescoBorzi** [comment]
@Meramaf thanks for your feedback, however it would be nice if more users can help us testing PRs **BEFORE** they get merged into master... so we prevent bad code to be merged.

**Pull request merged**